packages/nuxt/src/core/utils/parse.ts

```diff
@@ -0,0 +1,31 @@
+import { parseSync } from 'oxc-parser'
+import type { CatchClause, ClassBody, Declaration, Expression, MethodDefinition, ModuleDeclaration, ObjectProperty, Pattern, PrivateIdentifier, Program, PropertyDefinition, SpreadElement, Statement, Super, SwitchCase, TemplateElement } from 'oxc-parser'
+import { walk as _walk } from 'estree-walker'
+import type { SyncHandler } from 'estree-walker'
+import type { Node as ESTreeNode, Program as ESTreeProgram, ModuleSpecifier } from 'estree'
+
+/** estree also has AssignmentProperty, Identifier and Literal as possible node types */
+export type Node = Declaration | Expression | ClassBody | CatchClause | MethodDefinition | ModuleDeclaration | ModuleSpecifier | Pattern | PrivateIdentifier | Program | SpreadElement | Statement | Super | SwitchCase | TemplateElement | ObjectProperty | PropertyDefinition
+
+type InferThis<T extends (...args: any[]) => any> = T extends (this: infer U, ...args: infer A) => any ? U : unknown
```

Comment:
Can `InferThis` be replaced with `ThisParameterType` [handbook link](https://www.typescriptlang.org/docs/handbook/utility-types.html#thisparametertypetype)

Courtesy of [Dillon](https://bsky.app/profile/dillonis.online/post/3lbwtwbf4ck2b), thanks @dmmulroy

Reply:
perfect! TIL.